content/200-concepts/100-components/drivers.mdx

```diff
@@ -0,0 +1,134 @@
+metaTitle: 'Drivers (Preview)'
+metaDescription: 'Learn how you can use Prisma with other database drivers using driver adapters (Preview)'
+preview: true
+tocDepth: 3
+---
+
+## Built-in drivers (default)
+
+One of Prisma Client's component's is the [Query Engine](./prisma-engines/query-engine) <span class="concept"></span>. The Query Engine transforms Prisma Client queries to SQL statements that are executed using the included database driver. The Query Engine also includes built-in database drivers that don't require any additional effort to set up. The database drivers use TCP connections to connect to the database.
+
+![Query flow from the user application to the database with Prisma Client](./images/drivers/qe-query-execution-flow.png)
+
+## Driver adapters
+
+Prisma Client can connect and run queries against your database using other JavaScript database drivers using **driver adapters**. Adapters act as _translators_ between Prisma Client and the JavaScript database driver.
+
+![Query flow from the user application to the database using Prisma Client and driver adapters](./images//drivers/qe-query-engine-adapter.png)
+
+Prisma will use the Query Engine to transform the Prisma Client query to SQL and run the generated SQL queries via the JavaScript database driver.
+
+Prisma maintains the following driver adapters:
+
+- Serverless database driver adapters:
+  - [Neon](#neon)
+  - [PlanetScale](#planetscale)
+- Database driver adapters:
+  - [Turso](#turso)
+
+To use this feature:
+
+1. Update the `previewFeatures` block in your schema to include the the `driverAdapter` preview feature:
+
+   ```prisma
+   generator client {
+     provider        = "prisma-client-js"
+     previewFeatures = ["driverAdapters"]
+   }
+   ```
+
+2. Generate Prisma Client:
+
+   ```sh
+   npx prisma generate
+   ```
+
+### Serverless drivers
+
+By default, JavaScript database drivers use TCP connections. Several database providers enable use of different protocols, such as HTTP and WebSocket, to connect to your database. Example database providers include Neon, PlanetScale and Turso.
+
+#### Neon Serverless Driver
+
+Install the Prisma adapter for Neon, Neon serverless driver and `ws` packages:
+
+```sh
+npm install @prisma/adapter-neon @neondatabase/serverless ws
+```
+
+Update your Prisma Client instance:
+
+```ts
+import { Pool, neonConfig } from '@neondatabase/serverless'
+import { PrismaNeon } from '@prisma/adapter-neon'
+import { PrismaClient } from '@prisma/client'
+import dotenv from 'dotenv'
+import ws from 'ws'
+
+dotenv.config()
+neonConfig.webSocketConstructor = ws
+const connectionString = `${process.env.DATABASE_URL}`
+
+const pool = new Pool({ connectionString })
+const adapter = new PrismaNeon(pool)
+const prisma = new PrismaClient({ adapter })
+```
+
+We would appreciate your feedback on the Neon serverless driver adapter. Let us know what you think by leaving a comment in [this GitHub issue](https://github.com/prisma/prisma/discussions/21346).
```

Comment:
I think I would mention here (and for the other drivers) that they are in Preview / Early Access. 

Am I correct in understanding that Neon/PlanetScale are Preview while Turso is Early Access? @janpio what's the current state of that?

Reply:
Since all adapters are behind the `driverAdapters` Preview flag, I don't think we should make the distinction for Neon & PlanetScale. But I will add a note for Turso specifically because Prisma Migrate workflows don't work with it yet, while Prisma Migrate works just fine for Neon & PlanetScale.

Reply:
Correct differentiation, generally we talk about "driverAdapers" being in Preview, and specifically only Turso is EA.